src/Http/Http.Abstractions/src/IRouteHandlerFilter.cs

```diff
@@ -9,12 +9,12 @@ namespace Microsoft.AspNetCore.Http;
 public interface IRouteHandlerFilter
 {
     /// <summary>
-    /// Implements the core logic associated with the filter given a <see cref="RouteHandlerFilterContext"/>
+    /// Implements the core logic associated with the filter given a <see cref="RouteHandlerInvocationContext"/>
     /// and the next filter to call in the pipeline.
     /// </summary>
-    /// <param name="context">The <see cref="RouteHandlerFilterContext"/> associated with the current request/response.</param>
+    /// <param name="context">The <see cref="RouteHandlerInvocationContext"/> associated with the current request/response.</param>
     /// <param name="next">The next filter in the pipeline.</param>
     /// <returns>An awaitable result of calling the handler and apply
     /// any modifications made by filters in the pipeline.</returns>
-    ValueTask<object?> InvokeAsync(RouteHandlerFilterContext context, Func<RouteHandlerFilterContext, ValueTask<object?>> next);
+    ValueTask<object?> InvokeAsync(RouteHandlerInvocationContext context, RouteHandlerFilterDelegate next);
```

Comment:
How do we feel about adding a `RouteHandlerContext` parameter? We can try to approve this over email if we like it.

```suggestion
    ValueTask<object?> InvokeAsync(RouteHandlerContext routeContext, RouteHandlerInvocationContext invocationContext, RouteHandlerFilterDelegate next);
```

Reply:
Why? This should e a constructor argument, like middleware has the magic next no?

Reply:
How does middleware pass the magic `next`?

The implementation I just pushed passes the `RouteHandlerContext` as an argument to the object factory generated when the user utilizes the `AddFilter<TFilterType>()` overload.

Reply:
Follow up: https://github.com/dotnet/aspnetcore/issues/40724